### DISCUSSION TITLE
Duplicate values validation in FieldArray

### DISCUSSION THREAD

**Discussion by Joan-George** [react-hook-form/react-hook-form#10033] [Q&A]
How can I validate duplicate values in FieldArray?

This is my data
` [
{name:"test user",email:"testuser@gmail.com",role:["read_only"]},
{name:"test user",email:"testuser@gmail.com",role:["invoice_approver"]}
]`

I would like to validate the duplicate email.

**Comment by Sajarin-M** [comment]
You can use the `rules` prop in `useFieldArray` to achieve this, like:
```js
useFiledArray({
  rules: {
    validate: (arr) => {
      const emails = new Set(arr.map((obj) => obj.email));
      return emails.size === arr.length ? true : 'Duplicate email';
    },
  },
});
```
